File: RESTFUL_API/app/src/main/java/com/example/restful_api/MainActivity.java
package com.example.restful_api;

import android.os.Bundle;
import android.util.Log;
import androidx.appcompat.app.AppCompatActivity;

import com.example.restful_api.Service.BookApiService;
import com.example.restful_api.model.Book;

import java.util.List;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = MainActivity.class.getSimpleName();
    private BookApiService apiService;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        apiService =
            ApiClient.getRetrofitInstance().create(BookApiService.class);
// Contoh panggilan API
        getBooks();
    }
    private void getBooks() {
        Call<List<Book>> call = apiService.getBooks();
        call.enqueue(new Callback<List<Book>>() {
            @Override
            public void onResponse(Call<List<Book>> call,

                        Response<List<Book>> response) {

                if (response.isSuccessful()) {
                    List<Book> books = response.body();
                    for (Book book : books) {
                        Log.d(TAG, "Title: " + book.getTitle() +

                                ", Author: " + book.getAuthor());

                    }
                }
            }
            @Override
            public void onFailure(Call<List<Book>> call,

                Throwable t) {

                Log.e(TAG, "Error fetching books", t);
            }
        });
    }
// Implementasikan metode lain sesuai kebutuhan (create,update, delete)
}

File: UAS/app/src/main/java/com/example/uas/MainActivity.kt
package com.example.uas
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.lari.api_testing.databinding.ActivityMainBinding
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val userId = 1

        val apiService = RetrofitClient.apiService
        val call = apiService.getUser(userId)

        call.enqueue(object : Callback<User> {
            override fun onResponse(call: Call<User>, response: Response<User>) {
                if (response.isSuccessful) {
                    val user = response.body()
                    user?.let {
                        binding.id.text = user.id.toString()
                        binding.name.text = user.name
                        binding.birthDate.text = user.birthDate
                    }
                } else {
                    // Handle unsuccessful response
                    Toast.makeText(this@MainActivity, response.errorBody().toString(), Toast.LENGTH_SHORT).show()
                }
            }

            override fun onFailure(call: Call<User>, t: Throwable) {
                // Handle network or API call failure
                Toast.makeText(this@MainActivity, t.localizedMessage, Toast.LENGTH_SHORT).show()
            }
        })

    }
}